crates/egui/src/atomics/atomic_layout.rs

```diff
@@ -0,0 +1,392 @@
+        self.frame = frame;
+        self
+    }
+
+    /// Set the [`Sense`] used when allocating the [`Response`].
+    #[inline]
+    pub fn sense(mut self, sense: Sense) -> Self {
+        self.sense = sense;
+        self
+    }
+
+    /// Set the fallback (default) text color.
+    ///
+    /// Default: [`crate::Visuals::text_color`]
+    #[inline]
+    pub fn fallback_text_color(mut self, color: Color32) -> Self {
+        self.fallback_text_color = Some(color);
+        self
+    }
+
+    /// Set the minimum size of the Widget.
+    #[inline]
+    pub fn min_size(mut self, size: Vec2) -> Self {
+        self.min_size = size;
+        self
+    }
+
+    /// Set the [`Id`] used to allocate a [`Response`].
+    #[inline]
+    pub fn id(mut self, id: Id) -> Self {
+        self.id = Some(id);
+        self
+    }
+
+    /// Set the [`TextWrapMode`] for the [`Atomic`] marked as `shrink`.
+    ///
+    /// Only a single [`Atomic`] may shrink. If this (or `ui.wrap_mode()`) is not
+    /// [`TextWrapMode::Extend`] and no item is set to shrink, the first (right-most)
+    /// [`AtomicKind::Text`] will be set to shrink.
+    #[inline]
+    pub fn wrap_mode(mut self, wrap_mode: TextWrapMode) -> Self {
+        self.wrap_mode = Some(wrap_mode);
+        self
+    }
+
+    /// Set the [`Align2`].
+    ///
+    /// The default is chosen based on the [`Ui`]s [`crate::Layout`]. See
+    /// [this snapshot](https://github.com/emilk/egui/blob/master/tests/egui_tests/tests/snapshots/layout/button.png)
+    /// for info on how the [`crate::Layout`] affects the alignment.
+    #[inline]
+    pub fn align2(mut self, align2: Align2) -> Self {
+        self.align2 = Some(align2);
+        self
+    }
+
+    /// [`AtomicLayout::allocate`] and [`AllocatedAtomicLayout::paint`] in one go.
+    pub fn show(self, ui: &mut Ui) -> AtomicLayoutResponse {
+        self.allocate(ui).paint(ui)
+    }
+
+    /// Calculate sizes, create [`Galley`]s and allocate a [`Response`].
+    ///
+    /// Use the returned [`AllocatedAtomicLayout`] for painting.
+    pub fn allocate(self, ui: &mut Ui) -> AllocatedAtomicLayout<'a> {
+        let Self {
+            id,
+            mut atomics,
+            gap,
+            frame,
+            sense,
+            fallback_text_color,
+            min_size,
+            wrap_mode,
+            align2,
+        } = self;
+
+        let wrap_mode = wrap_mode.unwrap_or(ui.wrap_mode());
+
+        // If the TextWrapMode is not Extend, ensure there is some item marked as `shrink`.
+        // If none is found, mark the first text item as `shrink`.
+        if wrap_mode != TextWrapMode::Extend {
+            let any_shrink = atomics.iter().any(|a| a.shrink);
+            if !any_shrink {
+                let first_text = atomics
+                    .iter_mut()
+                    .find(|a| matches!(a.kind, AtomicKind::Text(..)));
+                if let Some(atomic) = first_text {
+                    atomic.shrink = true;
```

Comment:
What will this do for the text? Truncate?

```suggestion
                    atomic.shrink = true; // Will make the text truncate if needed
```

Reply:
It will wrap or truncate based on the wrap_mode. Since we check that wrap_mode != TextWrapMode::Extend it should always be able to shrink.